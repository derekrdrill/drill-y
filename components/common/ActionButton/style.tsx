import styled from 'styled-components';
import { Button } from '@mui/material';

export const ActionButtonRoot = styled(Button)<{
  $borderRadius?: number;
  $buttonColor?: string;
  $buttonHoverColor?: string;
  $buttonWidth?: string | number;
  $marginTop?: number;
  disabled?: boolean;
  variant?: string;
}>(
  ({
    $borderRadius,
    $buttonColor,
    $buttonHoverColor,
    $buttonWidth,
    $marginTop,
    disabled,
    variant,
  }) => ({
    '&:hover': {
      backgroundColor: `${$buttonHoverColor} !important`,
      color: `#FFFFFF !important`,
    },
    backgroundColor: `${variant === 'contained' ? `${$buttonColor}` : 'inherit'} !important`,
    borderColor: disabled
      ? 'gainsboro'
      : `${variant === 'outlined' ? `${$buttonColor}` : 'inherit'} !important`,
    borderRadius: `${$borderRadius ?? 75}px !important`,
    boxShadow: 'none !important',
    color: disabled
      ? 'gainsboro'
      : `${variant === 'outlined' ? `${$buttonColor}` : '#FFFFFF'} !important`,
    fontFamily: `'Josefin Sans', sans-serif !important`,
    fontWeight: '600 !important',
    fontSize: '12px !important',
    marginTop: `${$marginTop ?? 4}px !important`,
    paddingBottom: '4px !important',
    paddingLeft: '20px !important',
    paddingRight: '20px !important',
    paddingTop: '6px !important',
    width: $buttonWidth
      ? `${typeof $buttonWidth === `number` ? `${$buttonWidth}px` : $buttonWidth} !important`
      : 'inherit',
  }),
);
