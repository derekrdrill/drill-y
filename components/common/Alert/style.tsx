import styled from 'styled-components';
import { Grid } from '@mui/material';

// export const AlertContainer = styled(Grid)<{ isAlertHidden?: boolean; isAlertFading?: boolean }>(
//   ({ isAlertHidden, isAlertFading }) => ({
//     display: isAlertHidden && `none !important`,
//     opacity: isAlertHidden || isAlertFading ? 0 : 1,
//     position: 'fixed',
//     top: 125,
//     transition: 'all 350ms ease-in',
//     zIndex: 6,
//   }),
// );

export const AlertContainer = styled(Grid)<{
  $isAlertHidden?: boolean | undefined;
  $isAlertFading?: boolean | undefined;
}>(({ $isAlertHidden, $isAlertFading }) => ({
  display: `${$isAlertHidden ? 'none' : 'inline-flex'} !important`,
  opacity: $isAlertHidden || $isAlertFading ? 0 : 1,
  position: 'fixed',
  top: 125,
  transition: 'all 350ms ease-in',
  zIndex: 6,
}));
