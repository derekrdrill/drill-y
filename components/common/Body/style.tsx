import styled from 'styled-components';
import { Grid } from '@mui/material';

export const BodyRoot = styled(Grid)({
  zIndex: 0,
});
