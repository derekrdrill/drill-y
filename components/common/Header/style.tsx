import styled from 'styled-components';
import { AppBar, Grid } from '@mui/material';
import SearchIcon from '@mui/icons-material/Search';
import ShoppingBagIcon from '@mui/icons-material/ShoppingBag';

export const HeaderRoot = styled(AppBar)({
  backgroundColor: 'white !important',
  borderBottom: '1px solid #DFDFDF',
  boxShadow: 'none !important',
  zIndex: '3 !important',
});

export const HeaderTop = styled(Grid)({
  borderBottom: '1px solid #DFDFDF',
  padding: '5px 0',
});

export const HeaderLogo = styled.img({
  '&:hover': {
    opacity: 0.8,
  },
  cursor: 'pointer',
  height: 70,
  width: 100,
});

export const HeaderMenuText = styled.h4({
  '&:hover': {
    backgroundPositionX: '0%',
    backgroundSize: '100% 0.1em',
  },
  backgroundImage: 'linear-gradient(#212121, #212121)',
  backgroundPositionX: '100%',
  backgroundPositionY: '100%',
  backgroundRepeat: 'no-repeat',
  backgroundSize: '0% 0.1em',
  color: 'black',
  cursor: 'pointer',
  fontFamily: `'Josefin Sans', sans-serif !important`,
  fontWeight: '600 !important',
  textDecoration: 'none',
  textTransform: 'uppercase',
  transition: 'background-size 0.2s ease-in-out',
});

export const HeaderSearchIcon = styled(SearchIcon)({
  '&:hover': {
    color: '#a501c6',
    transform: 'scale(1.6)',
  },
  cursor: 'pointer',
  marginTop: 7,
  transition: 'all 5s ease-in-out',
  transform: 'scale(1.4)',
});
