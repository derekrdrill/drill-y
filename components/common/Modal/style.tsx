import styled from 'styled-components';
import { Grid } from '@mui/material';

export const ModalRootContainer = styled(Grid)({
  position: 'absolute',
  top: 200,
});

export const ModalContainer = styled(Grid)({
  backgroundColor: '#FFFFFF !important',
  borderRadius: 7,
  padding: '10px 20px',
});

export const ModalRow = styled(Grid)<{ isBottom?: boolean; isBody?: boolean }>(
  ({ isBottom, isBody }) => ({
    borderBottom: !isBottom ? '1px gainsboro solid' : 'none',
    margin: !isBottom ? '10px 0' : 0,
    padding: isBody ? '20px 0' : '5px 0',
  }),
);
