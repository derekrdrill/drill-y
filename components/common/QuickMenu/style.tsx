import styled from 'styled-components';
import { Grid } from '@mui/material';

export const QuickMenuRoot = styled(Grid)<{ $isQuickMenuOpen?: boolean }>(
  ({ $isQuickMenuOpen }) => ({
    backgroundColor: '#FFFFFF',
    boxShadow: '0 4px 2px -2px gray',
    left: 0,
    height: 400,
    minWidth: '100%',
    position: 'fixed',
    top: 123,
    transform: $isQuickMenuOpen ? 'none' : 'translateY(-105%)',
    transition: 'transform 300ms linear',
    width: '100%',
    zIndex: 2,
  }),
);
