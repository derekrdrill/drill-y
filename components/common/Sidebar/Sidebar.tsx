import * as React from 'react';
import { Grid } from '@mui/material';

import ActionButton from '../ActionButton/ActionButton';
import ComingSoon from '../../drill-y/ComingSoon/ComingSoon';
import Tooltip from '../Tooltip/Tooltip';

import { SidebarMenuLink, SidebarRoot, SidebarMenuText } from './style';

type SidebarProps = {
  isSidebarOpen: boolean;
  setIsSidebarOpen: React.Dispatch<React.SetStateAction<boolean>>;
};

const Sidebar = ({ isSidebarOpen, setIsSidebarOpen }: SidebarProps) => {
  return (
    <SidebarRoot display={{ xs: 'block', lg: 'none' }} $isSidebarOpen={isSidebarOpen}>
      <Grid container rowSpacing={2}>
        <Grid item xs={12}>
          <SidebarMenuLink href='/menu'>
            <SidebarMenuText>Menu</SidebarMenuText>
          </SidebarMenuLink>
        </Grid>
        <Tooltip tooltipTitle={<ComingSoon />} tooltipPlacement='right'>
          <Grid item xs={12}>
            <SidebarMenuText>Rewards</SidebarMenuText>
          </Grid>
        </Tooltip>
        <Tooltip tooltipTitle={<ComingSoon />} tooltipPlacement='right'>
          <Grid item xs={12}>
            <SidebarMenuText>Retail</SidebarMenuText>
          </Grid>
        </Tooltip>
        <Grid item xs={12}>
          <Grid container rowSpacing={2}>
            <Tooltip tooltipTitle={<ComingSoon />} tooltipPlacement='right'>
              <Grid item xs={12}>
                <ActionButton
                  buttonColor='#6a017f'
                  buttonHoverColor='#a501c6'
                  buttonWidth={'100%'}
                  size='large'
                  variant='contained'
                >
                  Join Rewards
                </ActionButton>
              </Grid>
            </Tooltip>
            <Tooltip tooltipTitle={<ComingSoon />} tooltipPlacement='right'>
              <Grid item xs={12}>
                <ActionButton
                  buttonColor='#6a017f'
                  buttonHoverColor='#a501c6'
                  buttonWidth={'100%'}
                  size='large'
                  variant='outlined'
                >
                  Sign In
                </ActionButton>
              </Grid>
            </Tooltip>
          </Grid>
        </Grid>
      </Grid>
    </SidebarRoot>
  );
};

export default Sidebar;
