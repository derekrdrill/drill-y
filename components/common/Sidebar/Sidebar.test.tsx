jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => jest.fn(),
}));

import '@testing-library/jest-dom';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import Sidebar from './Sidebar';

import { SidebarRoot } from './style';

describe('Sidebar unit tests', () => {
  it('renders as expected', () => {
    const result = renderer
      .create(<Sidebar isSidebarOpen={false} setIsSidebarOpen={jest.fn()} />)
      .toJSON();
    expect(result).toMatchSnapshot();
  });

  it('renders SidebarRoot styled component (open)', async () => {
    render(<SidebarRoot data-testid='menu-icon-container' $isSidebarOpen />);

    const sidebarRoot = await screen.findAllByTestId('menu-icon-container');
    expect(sidebarRoot[0]).toHaveClass('sc-gueYoa eNbQMd');
  });

  it('renders SidebarRoot styled component (not open)', async () => {
    render(<SidebarRoot data-testid='menu-icon-container' $isSidebarOpen={false} />);

    const sidebarRoot = await screen.findAllByTestId('menu-icon-container');
    expect(sidebarRoot[0]).toHaveClass('sc-gueYoa gGODpB');
  });
});
