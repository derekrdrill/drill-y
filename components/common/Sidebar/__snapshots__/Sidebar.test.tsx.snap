// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar unit tests renders as expected 1`] = `
<div
  className="MuiGrid-root sc-gueYoa gGODpB css-1ii0ocg-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-container css-1jzi3sl-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <h1
        className="sc-dmqHEX dkFvWg"
        onClick={[Function]}
      >
        Menu
      </h1>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <h1
        className="sc-dmqHEX dkFvWg"
      >
        Rewards
      </h1>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <h1
        className="sc-dmqHEX dkFvWg"
      >
        Retail
      </h1>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-container css-1jzi3sl-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium sc-beqWaB iSwvnH css-sghohy-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Join Rewards
          </button>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium sc-beqWaB iTzwqY css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Sign In
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
